lib/matplotlib/backend_bases.py

```diff
@@ -2241,6 +2191,68 @@ def print_figure(self, filename, dpi=None, facecolor=None, edgecolor=None,
             #self.figure.canvas.draw() ## seems superfluous
         return result
 
+    def tight_bbox(self, canvas, dpi, facecolor, edgecolor,
```

Comment:
(this is the PR I was looking for earlier). I think the current push is a pretty rough attempt at the `tight_bbox` method. It should be public, and probably ideally a method of `Figure`. This doesn't accomplish that, correct? I'm hesitant to include too many changes in this PR because this code is quite messy and a bit hard to follow